typescript/sdk/src/token/EvmERC20WarpRouteReader.ts

```diff
@@ -232,4 +234,28 @@ export class EvmERC20WarpRouteReader extends HyperlaneReader {
       ),
     );
   }
+
+  async fetchDestinationGas(
+    warpRouteAddress: Address,
+  ): Promise<DestinationGas> {
+    const warpRoute = TokenRouter__factory.connect(
+      warpRouteAddress,
+      this.provider,
+    );
+
+    /**
+     * @remark
+     * Router.domains() is used to enumerate the destination gas because GasRouter.destinationGas is not EnumerableMapExtended type
+     * This means that if a domain is removed, then we cannot read the destinationGas for it. This may impact updates.
+     */
+    const domains = await warpRoute.domains();
+
+    return Object.fromEntries(
+      await Promise.all(
+        domains.map(async (domain) => {
+          return [domain, (await warpRoute.destinationGas(domain)).toString()];
+        }),
+      ),
+    );
```

Comment:
I feel like this would be an excellent use case for the multicall discussion we had the other day